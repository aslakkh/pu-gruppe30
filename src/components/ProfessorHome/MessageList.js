/**
 * Created by jan on 21/03/2017.
 */
import React, { Component } from 'react'
import firebase from 'firebase';
import {ref } from '../../config/constants'
import {Form, FieldGroup, FormControl, ListGroup, ListGroupItem, Button} from 'react-bootstrap'


/*
* Displays list of messages and delete selected messages.
*/

export default class MessageList extends Component {
    constructor(props){
        super(props)
        this.state={
            courseID: this.props.courseID,
            messages:undefined
        }
    }

    getCommentsFromFirebase(){
        let messageRef = firebase.database().ref();
        let that = this;
        var messages =[]
        var koder = []
        messageRef.child('courses/'+this.state.courseID+'/Messages').on('value', function(snapshot) {
            snapshot.forEach(function(data){
                messages.unshift([ data.key,data.val().Message])

            })
            that.setState({
                messages:messages
            })
            messages=[]
    })
        console.log(this.state.messages)
    }

    componentWillMount(){
        this.getCommentsFromFirebase()
    }


handleClick(key) {
    var userRef = ref.child('courses/' + this.state.courseID + '/Messages/');
    userRef.child(key).remove();
}
    render(){

        if(this.state.messages === undefined || this.state.messages === null){
            return(
                <h5>You have no active courses. Add a course below.</h5>
            );
        }
        else{
            return(
                <div>

                    <ListGroup >
                        {Object.keys(this.state.messages).map((key,i) => {
                            return <ListGroupItem key={key} className="CoursesList">
                                {this.state.messages[key][1]}

                                <Button bsStyle="danger" onClick={() => this.handleClick(this.state.messages[key][0])}>Delete</Button>
                            </ListGroupItem>
                        })}
                    </ListGroup>
                </div>
            );
        }


    }
}