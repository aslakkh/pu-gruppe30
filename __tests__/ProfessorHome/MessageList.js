/**
 * Created by jan on 21/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import MessageList from '../../src/components/ProfessorHome/MessageList.js';
import TestUtils from "react-addons-test-utils"
import {mount} from 'enzyme'
import jest from 'jest'


describe("List Message", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <MessageList courseID={"TES1000"}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Not logged in", () => {
        expect(component2).toMatchSnapshot();
    });

    })


