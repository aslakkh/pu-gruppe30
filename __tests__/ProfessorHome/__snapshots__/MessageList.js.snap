// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Message Not logged in 1`] = `
<h5>
  No Messages
</h5>
`;
